### PR TITLE
feat: parse BIP-21 addresses in send modal

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Parses the URL field from BIP-21 URLs in UTXOs send modals.
Note, this does not yet parse amounts, which will require a bit more plumbing / architecting.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Contributes to https://github.com/shapeshift/web/issues/4206

## Risk

Current address/URL parsing may be borked. Confirmed EIP-681 / ENS / address parsing was still happy

## Testing

- When sending from an UTXO account/asset page, the address field from BIP-21 URLs are properly parsed, e.g:

- `bitcoin:someAddress`
- `litecoin:someAddress`
- `bitcoincash:someAddress`
- `bitcoincash:someAddress`

- If there are additional parameters e.g `amount`, the address is still correctly parsed. This can be tested by slapping `?amount=someAmount` after the examples above
- If there is a mismatch between the network of the address being sent to (e.g trying to send to a Litecoin BIP-21 URL from Dogecoin send), "Invalid address" is displayed
- Ensure this works with QR code scanning also (use a wallet that supports specifying amounts e.g Trust, and specify an amount to receive, which will ensure a BIP-21 QR code is generated)
- Current address / URL / name service parsing (e.g ENS/Y.at) shows no regressions

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Pasted Bitcoin BIP-21 URL on Bitcoin send modal (`'bitcoin:1BgGZ9tcN4rm9KBzDn7KprQz87SZ26SAMH?amount=20.3&label=Foobar`)

image
image

- Pasted Bitcoin Cash BIP-21 URL on BCH send modal:

image
image

- Pasted Litecoin BIP-21 URL on LTC send modal:

image
image

- Attempting to paste another UTXO URN in a given UTXO send modal:

image

Also tested with QR codes, though not posted because something something privacy.

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4216** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4216" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4216?utm_source=stack-comment).

File: src/lib/address/address.test.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+      expect(parseMaybeUrlByChainId(input)).toEqual(expectedOutput)
+    })
+
+    it('should not parse EIP-681 URL with parameters', () => {
+      const input = {
+        assetId: ethAssetId,
+        chainId: ethChainId,
+        value:
+          'ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD?value=2.014e18&gas=10&gasLimit=21000&gasPrice=50',
```
Comment (0xdef1cafe) [inline src/lib/address/address.test.ts:51]: getting some bizarre behavior pasting this address into the send modal, a weirdly rendered x character (copy pasted from in the input in the screenshot) ->`x`

Screenshot 2023-04-11 at 11 12 34 AM

Comment (0xdef1cafe) [reply src/lib/address/address.test.ts:51]: LLM rug?

Comment (gomesalexandre) [reply src/lib/address/address.test.ts:51]: Now that's correct, it's just the new font, here it is pasted here: `ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD?value=2.014e18&gas=10&gasLimit=21000&gasPrice=50`

The behavior is actually correct and so is the test. We should never parse EIP-681 with params as the test suggests. If you parse the l.34 one e.g `ethereum:0x1234DEADBEEF5678ABCD1234DEADBEEF5678ABCD` everything will be 🛍️ 
image

Comment (0xdef1cafe) [reply src/lib/address/address.test.ts:51]: hahaha fucking font rug. i checked with `charCodeAt` and it evaluated the same, so in that case, LGTM.

**Pull request merged**